strapi/strapi
# [DTS] Only delete types of data that are being transferred

### What does it do?

Update the import and transfer commands to generate the correct restore object

TODO:

- [ ] ~~refactor restore logic so it's modular and independent from local-source~~ I'm skipping this one because it would be a big structural change that might break things*

* I think we first need a discussion about if it would make more sense to make local-source a wrapper for a local-restore-source rather than local-source using strategies, since the strategies are/will be so tightly integrated it may be difficult to abstract them
 
### Why is it needed?

The initial functionality of the restore strategy causing a global reset of Strapi data caused too many users to be confused. 

The updated functionality of only touching data included in a transfer matches user expectations to not touch files that are excluded in a transfer or import.

### How to test it?

Using exclude or only with transfer or import should now only delete the files that are included in the process.

### Related issue(s)/PR(s)

Closes https://github.com/strapi/strapi/issues/17572

Completes DX-862, DX-863, DX-864
